api/v1alpha1/condition_consts.go

```diff
@@ -11,6 +11,28 @@ const (
 
 // Conditions and condition Reasons for the VirtualMachine object.
 
+const (
+	// VirtualMachineMetadataReadyCondition documents the state of the metadata passed to the VirtualMachine.
+	VirtualMachineMetadataReadyCondition ConditionType = "VirtualMachineMetadataReady"
+
+	// VirtualMachineMetadataDuplicatedReason (Severity=Error) documents that the mutually exclusive Secret and ConfigMap are
+	// specified in the VirtualMachineMetadata.
+	//
+	// This validation is performed by the validation webhook, defensively adding the reason.
+	VirtualMachineMetadataDuplicatedReason = "VirtualMachineMetadataDuplicated"
```

Comment:
Hmm.  Is there _any_ scenario where this will be hit though?  IIRC, we have liveness probes that fail the resource creation if the webhooks are down.